packages/dev/core/src/DeviceInput/webDeviceInputSystem.ts

```diff
@@ -411,7 +411,24 @@ export class WebDeviceInputSystem implements IDeviceInputSystem {
 
         this._pointerMoveEvent = (evt) => {
             const deviceType = this._getPointerType(evt);
-            const deviceSlot = deviceType === DeviceType.Mouse ? 0 : this._activeTouchIds.indexOf(evt.pointerId);
+            let deviceSlot = deviceType === DeviceType.Mouse ? 0 : this._activeTouchIds.indexOf(evt.pointerId);
+
+            // In the event that we're gettting pointermove events from touch inputs that we aren't tracking,
+            // look for an available slot and retroactively connect it.
+            if (deviceType === DeviceType.Touch && deviceSlot === -1) {

```

Comment:
how often does this code path happen as it sound potentially slow-er than the rest ?

Reply:
The code in this block should only happen if there's a touch move event that we aren't tracking (either because the max amount of touches have been exceeded or because we are getting move events from a touch device that we haven't recorded a down for).  If this happens to be an event that we didn't capture a down for beforehand, this block of code should only execute once in order to assign it a proper device slot.